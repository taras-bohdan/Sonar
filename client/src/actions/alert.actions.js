import { alertConstants } from '../constants';

export const alertActions = {
  success,
  error,
  clear,
};

/**
 * Create success alert action
 * @param {string} message - alert message
 * @returns {{type: string, message: *}} - success alert action with specified message
 */
function success(message) {
  return { type: alertConstants.SUCCESS, message };
}

/**
 * Create error alert action
 * @param {string} message - alert message
 * @returns {{type: string, message: *}} - error alert action with specified message
 */
function error(message) {
  return { type: alertConstants.ERROR, message };
}

/**
 * Create clear alert action
 * @returns {{type: string}} - clear alert action
 */
function clear() {
  return { type: alertConstants.CLEAR };
}
