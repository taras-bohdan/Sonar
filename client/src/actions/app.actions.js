import { appConstants } from '../constants';

/**
 * Togle drawer
 * @return {{type: string}} - toggle drawer action
 */
export function toggleDrawer() {
  return { type: appConstants.TOGGLE_DRAWER };
}
