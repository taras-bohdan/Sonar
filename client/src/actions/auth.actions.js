import { authConstants, userConstants } from '../constants';
import { userService } from '../services';

/**
 * Login success action
 * @return {{type: string}} - action
 */
export function loginSuccess() {
  return { type: authConstants.LOGIN_SUCCESS };
}

/**
 * Log out from app
 * @returns {{type: string}} - logout action
 */
export function logout() {
  userService.logout();
  return { type: userConstants.LOGOUT };
}
