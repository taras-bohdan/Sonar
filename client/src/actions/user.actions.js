import { userConstants } from '../constants';
import { userService } from '../services';
import { alertActions } from './';

export const userActions = {
  getAll,
};

/**
 * Get all users
 * @returns {Function} - get all action function
 */
function getAll() {
  return dispatch => {
    dispatch(request());

    userService.getAll()
      .then(
        response => dispatch(success(response.data)),
        error => {
          dispatch(failure(error));
          dispatch(alertActions.error(error.message));
        },
      );
  };


  /**
   * Create request action
   * @returns {{type: string}} - request processing action
   */
  function request() {
    return { type: userConstants.GETALL_REQUEST };
  }

  /**
   * Create success action
   * @param {object[]} users - users list
   * @returns {{type: string, users: *}} - request success action
   */
  function success(users) {
    return { type: userConstants.GETALL_SUCCESS, users };
  }

  /**
   * Create failure action
   * @param {object} error - error description
   * @returns {{type: string, error: *}} - request fail action
   */
  function failure(error) {
    return { type: userConstants.GETALL_FAILURE, error };
  }
}
