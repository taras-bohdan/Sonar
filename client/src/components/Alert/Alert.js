import React, { Component } from 'react';

import Snackbar from '@material-ui/core/Snackbar';
import SnackbarContent from '@material-ui/core/SnackbarContent';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import withStyles from '@material-ui/core/styles/withStyles';
import { connect } from 'react-redux';
import { func, object, string } from 'prop-types';

import { alertActions } from '../../actions/alert.actions';


const styles = theme => {
  return {
    close: {},
    error: {
      backgroundColor: theme.palette.error.dark,
    },
    success: {
      backgroundColor: theme.palette.primary.dark,
    },
  };
};

/**
 * Alert component
 * @return {*} - alert component
 */
export class Alert extends Component {
  handleClose = () => {
    const { dispatch } = this.props;
    dispatch(alertActions.clear());
  };

  /**
   * Render alert component
   * @return {*} - component's HTML
   */
  render() {
    const { classes, type, message } = this.props;
    return (
      <Snackbar
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'left',
        }}
        open={!!message}
        autoHideDuration={6000}
        onClose={this.handleClose}
      >
        <SnackbarContent
          className={classes[type]}
          message={<span id="message-id">{message}</span>}
          action={[
            <IconButton
              key="close"
              aria-label="Close"
              color="inherit"
              onClick={this.handleClose}
            >
              <CloseIcon/>
            </IconButton>,
          ]}>
        </SnackbarContent>
      </Snackbar>
    );
  }
}

/**
 * Map state to component's properties
 * @param {object} state - components state
 * @returns {{type: string, message: string}} - alert type and message
 */
function mapStateToProps(state) {
  const { type = 'success', message = '' } = state.alert;
  return {
    type,
    message,
  };
}

export default withStyles(styles, { withTheme: true })(connect(mapStateToProps)(Alert));

Alert.propTypes = {
  dispatch: func.isRequired,
  classes: object.isRequired,
  type: string.isRequired,
  message: string.isRequired,
};
