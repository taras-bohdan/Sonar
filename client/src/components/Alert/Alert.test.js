import React from 'react';
import { shallow, render } from 'enzyme';

import { Alert } from './Alert';
import { alertActions } from '../../actions';


/**
 * Get default component's props
 * @returns {object} - props
 */
function getDefaultProps() {
  return {
    classes: {},
    dispatch: jest.fn(),
    type: 'success',
    message: 'test message',
  };
}

describe('<Alert/>', () => {
  test('renders without error', () => {
    const props = getDefaultProps();
    shallow(<Alert {...props}/>);
  });

  test('Render successfully', () => {
    const props = getDefaultProps();

    const wrapper = render(
      <Alert {...props} />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('dispatch alert close action', () => {
    const props = getDefaultProps();
    const wrapper = shallow(<Alert {...props}/>);
    wrapper.instance().handleClose();
    expect(props.dispatch).toHaveBeenCalledWith(alertActions.clear());
  });

});

