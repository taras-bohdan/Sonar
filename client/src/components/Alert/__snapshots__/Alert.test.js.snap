// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert/> Render successfully 1`] = `
<div
  class="MuiSnackbar-root-1 MuiSnackbar-anchorOriginBottomLeft-7"
>
  <div
    class="MuiTypography-root-38 MuiTypography-body1-47 MuiPaper-root-11 MuiPaper-elevation6-19 MuiSnackbarContent-root-8"
    direction="up"
    role="alertdialog"
  >
    <div
      class="MuiSnackbarContent-message-9"
    >
      <span
        id="message-id"
      >
        test message
      </span>
    </div>
    <div
      class="MuiSnackbarContent-action-10"
    >
      <button
        aria-label="Close"
        class="MuiButtonBase-root-69 MuiIconButton-root-63 MuiIconButton-colorInherit-64"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label-68"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-72"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              >
                
              </path>
            </g>
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-79"
        >
          
        </span>
      </button>
    </div>
  </div>
</div>
`;
